build.gradle

```diff
@@ -1,7 +1,7 @@
 project(":solr-hadoop-common:solr-hadoop-document") {
 
     dependencies {
-        compile("org.apache.hadoop:hadoop-client:${hadoop2Version}") {
+        compile("org.apache.hadoop:hadoop-client:${hadoopVersion}") {
```

Comment:
Just thinking aloud here:

In theory, I like the flexibility that comes with having our `hadoopVersion` specified by our upstream "supermodule" (is that a word?).  But in practice, it's hard to take advantage of this flexibility.  Ideally, `hadoop-client` doesn't introduce any breaking change in the same major-version-line, but this does happen in practice.

Additionally, even when the Java APIs don't change, transitive dependencies sometimes do.  See some of the `LWSolrDocument` changes required by our implicit reliance on the Jackson deps brought in by `hadoop-client` as an example.  Though that can be solved with a more explicit gradle dependency list too.

Just wondering if the potential trappiness is worth the flexibility.